collector/container/rhtap-slim.Dockerfile

```diff
@@ -0,0 +1,151 @@
+ARG BUILD_DIR=/build
+ARG SRC_ROOT_DIR=${BUILD_DIR}
+ARG CMAKE_BUILD_DIR=${SRC_ROOT_DIR}/cmake-build
+ARG USE_VALGRIND=false
+ARG ADDRESS_SANITIZER=false
```

Comment:
These ARGs set here don't have any effect. IIUC, they are intended to be used in cmake configure step that is in the Makefile upstream 
https://github.com/stackrox/collector/blob/6d3c3822446be80f372bf3bc40e9cdabce7a6f61/collector/Makefile#L20-L28
(obviously the step cannot be reused here because there's no `docker` to exec).
These variables are introduced in a special file upstream - https://github.com/stackrox/collector/blob/6d3c3822446be80f372bf3bc40e9cdabce7a6f61/Makefile-constants.mk

The problem is that these ARGs don't become ENVs and so remain not visible to the actual command that's supposed to see them. Probably these specific don't cause issues because default values are also `false`.

I know the downstream dockerfile does this too and I know that it's not necessarily the part of this RHTAP onboarding to solve this problem, but...

The problem is that we have multiple differing patterns how these variables are set and used.
1. First is the upstream (see links above).
2. Downstream dockerfile that sets variables and [delegates](https://gitlab.cee.redhat.com/stackrox/rhacs-midstream/-/blob/rhacs-1.0-rhel-8/distgit/containers/rhacs-collector-slim/Dockerfile.in?ref_type=heads#L64) the rest to `build-collector.sh` file that [no longer exists](https://redhat-internal.slack.com/archives/CFMQ5C2TT/p1700843996467709).
3. This RHTAP dockerfile where you call the cmake command directly (`cmake -DDISABLE_PROFILING=ON -S "${SRC_ROOT_DIR}" -B "${CMAKE_BUILD_DIR}"`). No variables except for `DISABLE_PROFILING` are set. I'm not sure if the assumption that `cmake` respects environment variables is true. The main point is: this is different from the downstream dockerfile and from the upstream approach.

I find this to be a mess which needs fixing. Doing the same thing in three different ways is just too taxing on time and mental capacity. The situation with the downstream dockerfile after `build-collector.sh` removal needs to be addressed anyway. It would be good to have such a solution that will fit RHTAP dockerfile without much adaptation. I suggest getting in touch with the Collector team folks to find a solution.

Reply:
Updated PR with fixes from https://gitlab.cee.redhat.com/stackrox/rhacs-midstream/-/merge_requests/2278. 
ARGs are now set in the correct stages.